libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -203,12 +206,14 @@ string CHCSmtLib2Interface::querySolver(string const& _input)
 	util::h256 inputHash = util::keccak256(_input);
 	if (m_queryResponses.count(inputHash))
 		return m_queryResponses.at(inputHash);
-	if (m_smtCallback)
+
+	if ((m_enabledSolvers.smtlib2 || m_enabledSolvers.eld) && m_smtCallback)
```

Comment:
maybe unnecessary in the current version to test which solvers were enabled here

Reply:
If the callback is not allowed for other solvers, shouldn't we report an error instead of ignoring it?

Reply:
Actually, since this is CHC, we could add an assertion that the first arg of the conjunction should be true. What do you think @cameel ?